### PR TITLE
dbAuthCheck

### PR DESCRIPTION
**Pull request by denizul**
This provides a fix for the case when the API does not have credentials. The API would show that it was connected and provide incorrect log messages. 

Now the API will only proceed after having fully established authorisation. This is done by performing a DB Name search (which is only allowed for authorised users) for the target DB_NAME

### CODE REVIEW

File: API/models/base.go [hunk 1]
```diff
@@ -57,15 +58,39 @@ func init() {
...
+				found = true
+			}
+		}
+
+		//Target DB was not found
+		if found == false {
+			msg := "Requested Database was not found. Please check" +
+				"that your DB exists and that you are authorised to access it"
+			log.Fatal(msg)
+		}
```
Comment (GeremWD) [inline API/models/base.go:91]: We just log and we keep going ? I think an empty database with that name will be created if we do that.

File: API/models/base.go [hunk 2]
```diff
@@ -57,15 +58,39 @@ func init() {
...
+			log.Fatal("Authentication failed")
+		}
+		log.Fatal(e.Error())
+	} else {
+		found := false
+		for i := range dbs {
+			if dbs[i] == dbName {
+				found = true
+			}
+		}
```
Comment (GeremWD) [inline API/models/base.go:84]: There is an unused CheckIfDBExists that seems to be doing that.

Comment (denizul) [reply API/models/base.go]: I replaced it with the CheckIfDBExists function

File: API/models/base.go (hunk 1 above)
Comment (denizul) [reply API/models/base.go]: log.Fatal() is equivalent to println() followed by os.Exit(1)

**Review by GeremWD — APPROVED** [review]

**Pull request merged**